indico/modules/events/registration/schemas.py

```diff
@@ -33,3 +37,84 @@ def sort_list(self, data, many, **kwargs):
         if many:
             data = sorted(data, key=lambda tag: natural_sort_key(tag['title']))
         return data
+
+
+# Schemas for the Check-in app API
+
+class CheckinEventSchema(mm.SQLAlchemyAutoSchema):
+    class Meta:
+        model = Event
+        fields = ('event_id', 'title', 'description', 'start_dt', 'end_dt')
+
+    event_id = fields.Int(attribute='id')
+
+
+class CheckinRegFormSchema(mm.SQLAlchemyAutoSchema):
+    class Meta:
+        model = RegistrationForm
+        fields = ('regform_id', 'title', 'introduction', 'start_dt', 'end_dt',
+                  'registration_count', 'checked_in_count')
+
+    regform_id = fields.Int(attribute='id')
+    registration_count = fields.Function(lambda regform: len(regform.registrations))
+    checked_in_count = fields.Function(lambda regform: len(regform.checked_in_registrations))
```

Comment:
This is quite expensive for regforms with thousands of registrations. You query them twice...

for the total count see if `active_registration_count` or `existing_registrations_count` do what you want, and you could add another column property for the checked-in ones